packages/ember-views/lib/system/build-component-template.js

```diff
@@ -186,6 +211,25 @@ function normalizeComponentAttributes(component, isAngleBracket, attrs) {
     }
   }
 
+  // IE8 Support: IE8 cannot change the type attr of an input after it has been appended to
+  // any node. Therefore, we detect if the browser cannot change the type of the input after
+  // being appended, and unbind type.
+  if (normalized.type && !canChangeType) {
```

Comment:
Simpler is much less likely to cause other issues down the line.  We should ensure that a deprecation is printed when you use IE8 (I vaguely think that there is already), and remove the extra complication here.